server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -915,6 +920,8 @@ private void processBulkIndexIngestRequest(
             original.numberOfActions(),
             () -> bulkRequestModifier,
             bulkRequestModifier::markItemAsDropped,
+            (indexName) -> shouldStoreFailure(indexName, metadata, System.currentTimeMillis()),
```

Comment:
Hmm... since we're going to potentially make this call for each failed doc, and that could be a large number of docs, should this use `threadPool.absoluteTimeInMillis()` instead of `System.currentTimeMillis()`, since the former is cached?